libsolidity/interface/CompilerStack.cpp

```diff
@@ -1921,3 +1949,13 @@ Json::Value CompilerStack::gasEstimates(std::string const& _contractName) const
 
 	return output;
 }
+
+bool CompilerStack::isAnySourceExperimentalSolidity() const
```

Comment:
Of course, if you do end up switching to the `front()` check for experimental Solidity, might as well change the name of this function to `isExperimentalSolidity()`.